### PR TITLE
Task6: Change detection

### PR DESCRIPTION
**Pull request by MaryBozhyk**

### CODE REVIEW

File: src/app/courses-page/services/courses.service.ts [hunk 1]
```diff
@@ -41,8 +41,15 @@ export class CoursesService {
     return this.courses;
   }
 
-  createCourse(course: Course): void {
-    this.courses.push(course);
+  createCourse(course: Partial<Course>): void {
+    const updatedCourse = {
+      id: this.courses[this.courses.length - 1].id + 1,
```
Comment (SteveRusin) [inline src/app/courses-page/services/courses.service.ts]: do not rely on array length, you will end up with two same ids if you add a course, delete some other and add new one
here is the package for id generation, you can use this)
https://www.npmjs.com/package/uuid

In idead world id should be assigned by backend

File: src/app/new-courses-page/course-date/course-date.component.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  templateUrl: './course-date.component.html',
+  styleUrls: ['./course-date.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class CourseDateComponent {
+  @Input() courseDate: Date;
+  @Output() courseDateChange = new EventEmitter<Date>();
+
+  onDateChange(model: Date): void {
+    this.courseDate = model;
```
Comment (SteveRusin) [inline src/app/new-courses-page/course-date/course-date.component.ts]: Do not change the input. It should be changed by the binding in the parent component. That's called unidirectional data flow
https://blog.angular-university.io/angular-2-what-is-unidirectional-data-flow-development-mode/

File: src/app/new-courses-page/course-duration/course-duration.component.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  templateUrl: './course-duration.component.html',
+  styleUrls: ['./course-duration.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class CourseDurationComponent {
+  @Input() courseDuration: number;
+  @Output() courseDurationChange = new EventEmitter<number>();
+
+  onDurationChange(model: number): void {
+    this.courseDuration = model;
```
Comment (SteveRusin) [inline src/app/new-courses-page/course-duration/course-duration.component.ts]: same here)

File: src/app/new-courses-page/new-course-form/new-course-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Output } from '@angular/core';
+
+import { Course } from '@app/models';
+
+@Component({
+  selector: 'app-new-course-form',
+  templateUrl: './new-course-form.component.html',
+  styleUrls: ['./new-course-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
```
Comment (SteveRusin) [inline src/app/new-courses-page/new-course-form/new-course-form.component.ts]: you don't have any inputs here. I don't think it makes sense to use onPush here.
Although, on push not only about inputs, it still better to use it when you have inputs and they change.

File: src/app/courses-page/services/courses.service.ts (hunk 1 above)
Comment (MaryBozhyk) [reply src/app/courses-page/services/courses.service.ts]: @SteveRusin uuid is used. Thanks)

File: src/app/new-courses-page/course-date/course-date.component.ts (hunk 2 above)
Comment (MaryBozhyk) [reply src/app/new-courses-page/course-date/course-date.component.ts]: Changed. Thanks)

File: src/app/new-courses-page/course-duration/course-duration.component.ts (hunk 3 above)
Comment (MaryBozhyk) [reply src/app/new-courses-page/course-duration/course-duration.component.ts]: Changed. Thanks)

File: src/app/new-courses-page/new-course-form/new-course-form.component.ts (hunk 4 above)
Comment (MaryBozhyk) [reply src/app/new-courses-page/new-course-form/new-course-form.component.ts]: removed
thanks)

**Pull request merged**